extension/src/websocket-forwarder.ts

```diff
@@ -35,16 +35,26 @@ export class WebSocketForwarder implements vscode.Disposable {
     });
   }
 
-  private handleClientMessage(message: string) {
-    const obj = JSON.parse(message);
-    if (obj?.method === 'action') {
-      const handler = this.actionHandlerFor(obj.params);
+  private handleClientMessage(message: unknown) {
+    if (this.isAction(message)) {
```

Comment:
@ivy-lli feels better, but I can't prove whether it's actually better

Reply:
Looks good